File: Visual Studio LIghtswitch 2012/Chapter15/HelpDeskVB/HelpDeskVB/Server/MailService.svc.vb
'You can use and redistribute the code from this book (and sample application) for non-commercial and 
'most commercial purposes as long as you acknowledge the source and authorship. 
'You should note the source of the code in any documentation as well as in the program code itself (as a comment). 
'The acknowledgment should include author, title, publisher, and ISBN. 
'An example of such an acknowledgment would be "Derived from Listing 2-10, LightSwitch 2012 by Tim Leung, published by Apress, ISBN 9781430250715".

' NOTE: You can use the "Rename" command on the context menu to change the class name "MailService" in code, svc and config file together.
' NOTE: In order to launch WCF Test Client for testing this service, please select MailService.svc or MailService.svc.vb at the Solution Explorer and start debugging.
Imports System.ServiceModel.Activation

'Listing 15-17. Implementing Your Web Service Method
<AspNetCompatibilityRequirements(
    RequirementsMode:=AspNetCompatibilityRequirementsMode.Required)> _
Public Class MailService
    Implements IMailService

    Public Function SendMail(
       emailTo As String, subject As String, body As String
          ) As String Implements IMailService.SendMail
        Try
            Dim senderEmail As String = "admin@lsfaq.com"
            SmtpMailHelper.SendMail(
               senderEmail, emailTo, subject, body,
               Nothing, Nothing) ' optional - add email attachment here...
            Return ("Email Sent")
        Catch ex As Exception
            Return ex.Message
        End Try
    End Function

End Class

File: Visual Studio LIghtswitch 2012/Chapter15/HelpDeskCS/HelpDeskCS/Server/MailService.svc.cs
//You can use and redistribute the code from this book (and sample application) for non-commercial and 
//most commercial purposes as long as you acknowledge the source and authorship. 
//You should note the source of the code in any documentation as well as in the program code itself (as a comment). 
//The acknowledgment should include author, title, publisher, and ISBN. 
//An example of such an acknowledgment would be "Derived from Listing 2-10, LightSwitch 2012 by Tim Leung, published by Apress, ISBN 9781430250715".

using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.Serialization;
using System.ServiceModel;
using System.Text;
using System.ServiceModel.Activation;

//Listing 15-17. Implementing Your Web Service Method
namespace LightSwitchApplication
{
    [AspNetCompatibilityRequirements(
        RequirementsMode = AspNetCompatibilityRequirementsMode.Allowed)]
    public class MailService : IMailService
    {
        public string SendMail(
           string emailTo, string subject, string body)
        {
            try
            {
                string senderEmail = "admin@lsfaq.com";
                LightSwitchApplication.UserCode.SmtpMailHelper.SendMail(
                   senderEmail, emailTo, subject, body,
                   null, null); // optional - add email attachment here...
                return ("Email Sent");
            }
            catch (Exception ex)
            {
                return ex.Message;
            }
        }
    }
}
